README.md

```diff
@@ -73,24 +56,221 @@ The `-v /path/to/your/local/models:/workspace/models` part will make sure
+    image_loc=DataLocation('filesystem', '/path/to/image1.jpg'),
+    # Where the feature vector should be output to.
+    # CoralNet uses a custom .featurevector extension for these, but the
+    # format is just JSON.
+    feature_loc=DataLocation('filesystem', '/path/to/image1.featurevector'),
+)
+return_message = extract_features(message)
+print("Feature vector stored at: /path/to/image1.featurevector")
+print(f"Runtime: {return_message.runtime}")
+```
+
+### train_classifier
+
+Takes:
+
+- Feature vectors, each vector corresponding to a set of pixel locations in one image
+- Ground-truth (typically human-confirmed) annotations corresponding to those feature vectors
+- Optionally, previously-created classifiers to re-evaluate with these annotations
+- Training parameters
+
+Produces a classifier (model) loadable in scikit-learn, and classifier evaluation results. Example:
+
+```python
+from spacer.data_classes import ImageLabels
+from spacer.messages import DataLocation, TrainClassifierMsg
+from spacer.tasks import train_classifier
+
+message = TrainClassifierMsg(
+    # Identifier to set this training job apart from others. Should be
+    # unique for each classifier.
+    job_token='classifier1',
+    # 'minibatch' is currently the only trainer available.
+    trainer_name='minibatch',
+    # How many iterations the training algorithm should run; more epochs
+    # = more opportunity to converge to a better fit, but slower.
+    nbr_epochs=10,
+    # Classifier types available:
+    # 1. 'MLP': multi-layer perceptron; newer classifier type for CoralNet
+    # 2. 'LR': logistic regression; older classifier type for CoralNet
+    clf_type='MLP',
+    # Point-locations to ground-truth-labels (annotations) mapping. Used for
+    # training the classifier.
+    # The data dict-keys must be the same as the `key` used in the
+    # extract-features task's `feature_loc`.
+    # The data dict-values are lists of tuples of
+    # (row, column, label ID). You'll need to be tracking a mapping of
+    # integer label IDs to the labels you use.
+    train_labels=ImageLabels(data={
+        '/path/to/image1.featurevector': [(1000, 2000, 1), (3000, 2000, 2)],
+        '/path/to/image2.featurevector': [(1000, 2000, 3), (3000, 2000, 1)],
+        '/path/to/image3.featurevector': [(1234, 2857, 11), (3094, 2262, 25)],
+    }),
+    # Point-locations to ground-truth-labels mapping. Used for evaluating
+    # the classifier's accuracy. Should be disjoint from train_labels.
+    # CoralNet uses a 7-to-1 ratio of train_labels to val_labels.
+    val_labels=ImageLabels(data={
+        '/path/to/image4.featurevector': [(500, 2500, 1), (2500, 1500, 3)],
+        '/path/to/image5.featurevector': [(4321, 5582, 25), (4903, 2622, 19)],
+    }),
+    # Partially-filled-in DataLocation which specifies the storage type
+    # and S3 bucket (if applicable) that all the feature vectors are
+    # stored at, but doesn't specify a key.
```

Comment:
I'm not understanding this last point about not specifying a key

Reply:
Would this clarify better?: "This DataLocation should specify the storage_type arg, and the S3 bucket_name arg if applicable, corresponding to where the feature vectors are stored at. But the key arg isn't used."

This also implies a requirement that should get documented: all the train_labels, val_labels, and probably ref_labels later must live under the same storage_type / bucket_name.

Reply:
Yes that sounds good -- maybe `The key arg isn't used, because the location is for all feature vectors, not any particular key."

Reply:
See commit 83b70f6, trying different wording: "All the feature vectors should use the same storage_type, and the same S3 bucket_name if applicable. This DataLocation's purpose is to describe those common storage details. The key arg is ignored, because that will be different for each feature vector."